app/controllers/page_mount_controller.rb

```diff
@@ -0,0 +1,126 @@
+class PageMountController < SiteController
+  no_login_required
+
+  radiant_layout { |controller| controller.domains_layout }
+
+  before_filter :set_monitoring_meta_data, only: :proxy
+
+  # Example:
+  #  /accommodation/kangaroo-island/ozone-12?adults=1
+  #
+  # params[:paths]  => ["kangaroo-island", "ozone-12"]
+  # params[:adults] => 1
+  def proxy
```

Comment:
While I understand that this is a pure extraction, I also feel it necessary to point out that this method could really be split into private methods / classes extractions.